pkg/policy/api/selector.go

```diff
@@ -146,6 +153,28 @@ func (n EndpointSelector) GetMatch(key string) ([]string, bool) {
 	return nil, false
 }
 
+// labelSelectorToRequirements turns a kubernetes Selector into a slice of
+// requirements equivalent to the selector. These are cached internally in the
+// EndpointSelector to speed up Matches().
+//
+// This validates the labels, which can be expensive (and may fail..)
+// If there's an error, the selector will be nil and the Matches()
+// implementation will refuse to match any labels.
+func labelSelectorToRequirements(labelSelector *metav1.LabelSelector) *k8sLbls.Requirements {
+	selector, err := metav1.LabelSelectorAsSelector(labelSelector)
+	if err != nil {
```

Comment:
We can this fail in practice?

Reply:
When looking at the source code, the label key and value could fail validation from the K8s side, i.e. if the length of the value of the label is too long, or it doesn't comply with some specified regexes in K8s. Are you asking whether we even need to care about the result of the error if we have gotten to this point (i.e., validation has already been performed on the labels at import stage and thus we would never fail at this point in the API?). 

Such validation of the labels may occur on the K8s side (would be interesting to see if this is the case), but I don't think we validate the labels themselves within Cilium. We do no sanitization of the `FromEndpoints` or `ToEndpoints` provided in a CNP:

https://github.com/cilium/cilium/blob/9aa9c768baef69b07598340ac57ab632d225b976/pkg/policy/api/rule_validation.go#L56-L119

Perhaps we could validate the key and value provided in the CNP conform with K8s restrictions, and then we can ignore the error here as we can be assured that the LabelSelectors have been validated already.

Reply:
Given we can fall back, what we have seems fine. Mostly I wanted to understand when this would likely fail and we would see a performance degradation again.

Reply:
Can you clarify what you mean by "fall back" - to what exactly? We need to convert the LabelSelector to a Selector to be able to do a match either way, so we could still fail in the conversion process and not be able to do a match.

Reply:
@tgraf do you want me to validate the labels that are provided in FromEndpoints / ToEndpoints as part of our Sanitization in the API? I think if we do that, we can ignore the error while converting them to labelSelectors when we are performing match operations. I can explore this / create an issue for it if you like.

Reply:
Please create an issue. It is definitely preferred to do validation as we can surface the error in the CNP status.

Reply:
Finally got around to filing issue - marked for 1.3-bugfix milestone. #5336